pkg/controller/appdefinition/deploy.go

```diff
@@ -627,7 +626,11 @@ func ToDeployments(req router.Request, appInstance *v1.AppInstance, tag name.Ref
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, dep)
+		sa := toServiceAccount(dep.GetName(), dep.GetLabels(), dep.GetAnnotations(), appInstance)
+		if perms := v1.FindPermission(dep.GetName(), appInstance.Spec.Permissions); perms.HasRules() {
+			result = append(result, toPermissions(perms, dep.GetLabels(), dep.GetAnnotations(), appInstance)...)
+		}
+		result = append(result, dep, sa)
```

Comment:
For those coming back to review, this is how I'm doing the new logic that always creates a ServiceAccount and then checks to see if it should create Roles/Bindings.